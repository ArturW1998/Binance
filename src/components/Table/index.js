import React from 'react';
import PropTypes from 'prop-types';
import { LISTS } from '../../constants';
import Row from '../Row';

const { TABLE_DATA_LIST, PAIRS_LIST } = LISTS;
const BTC = PAIRS_LIST[1];

const Table = ({ ticker, filter }) => {
  const tickerArray = Object.values(ticker);

  return (
    <ul className="table">
      <li className="d-none d-md-inline">
        <div className="row table-header small py-1">
          {TABLE_DATA_LIST.map(({ title }) => (
            <div key={title} className="col">
              <h2 className="title font-weight-bold mb-0">{title}</h2>
            </div>
          ))}
        </div>
      </li>

      {tickerArray.map(row =>
        row.symbol.endsWith(filter) ? <Row key={row.symbol} {...row} /> : null,
      )}
    </ul>
  );
};

Table.propTypes = {
  ticker: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
  filter: PropTypes.string,
};

Table.defaultProps = {
  ticker: {},
  filter: BTC,
};

export default Table;
