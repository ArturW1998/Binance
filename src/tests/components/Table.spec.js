import React from 'react';
import { shallow } from 'enzyme';
import Table from '../../components/Table';

describe('Table component', () => {
  describe('Table component with filter and ticker', () => {
    const props = {
      filter: 'BTC',
      ticker: {
        BNBBTC: {
          symbol: 'BNBBTC',
        },
        ETHUSDT: {
          symbol: 'ETHUSDT',
        },
        ENJBTC: {
          symbol: 'ENJBTC',
        },
      },
    };

    const table = shallow(<Table {...props} />);

    it('renders properly', () => {
      expect(table).toMatchSnapshot();
    });

    it('renders 2 rows', () => {
      expect(table.find('Row')).toHaveLength(2);
    });

    describe('correct render for first row', () => {
      it('Pair', () => {
        expect(
          table
            .find('h2')
            .first()
            .text(),
        ).toEqual('Pair');
      });

      it('info', () => {
        expect(
          table
            .find('h2')
            .at(1)
            .text(),
        ).toEqual('Latest Price');
      });
    });
  });

  describe('Table component with default props', () => {
    const props = {};

    const table = shallow(<Table {...props} />);

    it('renders properly', () => {
      expect(table).toMatchSnapshot();
    });

    it('not render rows', () => {
      expect(table.find('Row')).toHaveLength(0);
    });
  });
});
