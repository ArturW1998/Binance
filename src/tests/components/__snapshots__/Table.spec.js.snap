// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Table component with default props renders properly 1`] = `ShallowWrapper {}`;

exports[`Table component Table component with filter and ticker renders properly 1`] = `ShallowWrapper {}`;
